src/NServiceBus.RabbitMQ/RabbitMQTransportInfrastructure.cs

```diff
@@ -30,19 +31,42 @@ public RabbitMQTransportInfrastructure(SettingsHolder settings, string connectio
             connectionFactory = new ConnectionFactory(connectionConfiguration, clientCertificates);
 
             routingTopology = CreateRoutingTopology();
+            settings.Set(SettingsKeys.RoutingTopologySupportsNativeDelays, routingTopology is ISupportNativeDelays);
 
             bool usePublisherConfirms;
             if (!settings.TryGet(SettingsKeys.UsePublisherConfirms, out usePublisherConfirms))
             {
                 usePublisherConfirms = true;
             }
 
-            channelProvider = new ChannelProvider(connectionFactory, routingTopology, usePublisherConfirms);
+            settings.TryGet(SettingsKeys.DisableTimeoutManager, out disableTimeoutManager);
+
+            bool allEndpointsSupportNativeDelays;
+            settings.TryGet(SettingsKeys.AllEndpointsSupportNativeDelays, out allEndpointsSupportNativeDelays);
+
+            channelProvider = new ChannelProvider(connectionFactory, routingTopology, usePublisherConfirms, allEndpointsSupportNativeDelays);
 
             RequireOutboxConsent = false;
         }
 
-        public override IEnumerable<Type> DeliveryConstraints => new[] { typeof(DiscardIfNotReceivedBefore), typeof(NonDurableDelivery) };
+        public override IEnumerable<Type> DeliveryConstraints
+        {
+            get
+            {
+                var constraints = new List<Type>
+                {
+                    typeof(DiscardIfNotReceivedBefore),
+                    typeof(NonDurableDelivery)
+                };
+
+                if (disableTimeoutManager)
+                {
+                    constraints.Add(typeof(DelayedDelivery.DelayedDeliveryConstraint));
```

Comment:
> it probably would. Is that achievable in a nice way?

Yes, just need to add `DoNotDeliverBefore` and `DelayDeliveryWith` to the `constraints` collection instead of `DelayedDeliveryConstraint`.